File: solutions/go/darts/1/darts.go
package darts

import "math"

func Score(x, y float64) int {
	pos := (math.Abs(x) + math.Abs(y)) / 2.0
	if pos > 7 {
		return 0
	} else if pos > 3.5 {
		return 1
	} else if pos <= 3.5 && pos > 0.7 {
		return 5
	} else if pos <= 0.7 {
		return 10
	}
	return 0
}

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"
#include <cmath>

namespace darts {

double score(double x, double y) {
  double tmp = std::abs(x) + std::abs(y);
  tmp /= 2.0;
  if (tmp > 7.0) {
    return 0.0;
  } else if (tmp > 3.5 && tmp <= 7.0) {
    return 1.0;
  } else if (tmp <= 3.5 && tmp > 0.7) {
    return 5.0;
  } else if (tmp <= 0.7) {
    return 10;
  }
  return 0.0;
}

} // namespace darts